### PR TITLE
[#3885] feat(catalog-lakehouse-paimon): Add a basic code skeleton for Paimon catalog

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?

Based on https://github.com/datastrato/gravitino/pull/2746 contributed by @SteNicholas.

Add a basic code skeleton for Paimon catalog.

### Why are the changes needed?

Fix: https://github.com/datastrato/gravitino/issues/3885

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Just build.

### CODE REVIEW

**Comment by caican00** [comment]
Hi @FANNG1 @SteNicholas , could you help review this pr? Thanks!

File: catalogs/catalog-lakehouse-paimon/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    exclude("com.zaxxer", "HikariCP")
+    exclude("com.sun.jersey", "jersey-server")
+  }
+
+  annotationProcessor(libs.lombok)
+  compileOnly(libs.lombok)
+
+  testImplementation(libs.junit.jupiter.api)
+  testImplementation(libs.junit.jupiter.params)
+  testImplementation(libs.mockito.core)
```
[the remark this answers is not in the document]
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/build.gradle.kts:71]: yes, i will remove them.

[the remark this answers is not in the document]
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/build.gradle.kts]: updated. cc @FANNG1

File: catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/PaimonSchema.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/PaimonSchema.java:2]: please correct all `Copyright 2023`

**Comment by FANNG1** [comment]
LGTM, except one comment

File: catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/PaimonSchema.java (hunk 2 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-paimon/src/main/java/com/datastrato/gravitino/catalog/lakehouse/paimon/PaimonSchema.java]: Fixed.

**Comment by caican00** [comment]
Hi @FANNG1 , could you help to add @SteNicholas  as a co-author?

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
> Hi @FANNG1 , could you help to add @SteNicholas as a co-author?

done

**Pull request merged**